geoalchemy2/admin/dialects/sqlite.py

```diff
@@ -209,15 +395,55 @@ def after_create(table, bind, **kw):
 def before_drop(table, bind, **kw):
     """Handle spatial indexes during the before_drop event."""
     dialect, gis_cols, regular_cols = setup_create_drop(table, bind)
-    for col in gis_cols:
-        # Disable spatial indexes if present
-        disable_spatial_index(bind, table, col)
+    db_is_gpkg = is_gpkg(bind)
 
-        args = [table.name, col.name]
+    if not db_is_gpkg:
+        for col in gis_cols:
+            # Disable spatial indexes if present
+            disable_spatial_index(bind, table, col)
 
-        stmt = select(*_format_select_args(func.DiscardGeometryColumn(*args)))
-        stmt = stmt.execution_options(autocommit=True)
-        bind.execute(stmt)
+            args = [table.name, col.name]
+
+            stmt = select(*_format_select_args(func.DiscardGeometryColumn(*args)))
+            stmt = stmt.execution_options(autocommit=True)
+            bind.execute(stmt)
+    else:
+        for col in gis_cols:
+            # Disable spatial indexes if present
+            # TODO: This is useless but if we remove it then the disable_spatial_index should be
+            # tested separately
+            disable_spatial_index(bind, table, col)
+
+            # Remove metadata from internal tables
+            # (this is equivalent to DiscardGeometryColumn but for GeoPackage)
+            bind.execute(
+                text(
+                    """DELETE FROM gpkg_extensions
+                    WHERE table_name = '{}'
```

Comment:
Good to add `lower()` around both sides of the equality, like here:
https://github.com/OSGeo/gdal/blob/6e9103bd5acb1ff8da305c4e77fa30335ffa3b70/ogr/ogrsf_frmts/gpkg/ogrgeopackagedatasource.cpp#L6644

Reply:
:+1: